packages/components/src/components/SquiggleViewer/ExpressionViewer.tsx

```diff
@@ -42,131 +46,133 @@ const VariableList: React.FC<{
-        <VariableBox
-          value={value}
-          heading={`Distribution (${distType})\n${
-            distType === SqDistributionTag.Symbolic
-              ? value.value.toString()
-              : ""
-          }`}
-          renderSettingsMenu={({ onChange }) => {
-            const shape = value.path
-              ? value.value.pointSet(value.path.project.getEnvironment())
-              : undefined;
+      return {
+        value: value,
+        heading: `${distType} Distribution`,
+        renderSettingsMenu: ({ onChange }) => {
+          const shape = value.path
+            ? value.value.pointSet(value.path.project.getEnvironment())
+            : undefined;
 
-            return (
-              <ItemSettingsMenu
-                value={value}
-                onChange={onChange}
-                metaSettings={{
-                  disableLogX:
-                    shape?.ok && hasMassBelowZero(shape.value.asShape()),
-                }}
-                withFunctionSettings={false}
-              />
-            );
-          }}
-        >
-          {(settings) => {
-            const plot = SqDistributionsPlot.create({
-              distribution: value.value,
-              ...generateDistributionPlotSettings(
-                settings.distributionChartSettings
-              ),
-            });
+          return (
+            <ItemSettingsMenu
+              value={value}
+              onChange={onChange}
+              metaSettings={{
+                disableLogX:
+                  shape?.ok && hasMassBelowZero(shape.value.asShape()),
+              }}
+              withFunctionSettings={false}
+            />
+          );
+        },
+        children: (settings) => {
+          const plot = SqDistributionsPlot.create({
+            distribution: value.value,
+            ...generateDistributionPlotSettings(
+              settings.distributionChartSettings
+            ),
+          });
 
-            return (
-              <DistributionsChart
-                plot={plot}
-                environment={environment}
-                height={settings.chartHeight}
-              />
-            );
-          }}
-        </VariableBox>
-      );
+          return (
+            <DistributionsChart
+              plot={plot}
+              environment={environment}
+              height={settings.chartHeight}
+            />
+          );
+        },
+      };
     }
     case "String":
-      return (
-        <VariableBox value={value} heading="String">
-          {() => (
-            <>
-              <span className="text-neutral-300">"</span>
-              <span className="text-neutral-600 text-sm">{value.value}</span>
-              <span className="text-neutral-300">"</span>
-            </>
-          )}
-        </VariableBox>
-      );
+      return {
+        value: value,
+        heading: "String",
+        preview:
+          value.value.substring(0, 30) + (value.value.length > 30 ? "..." : ""),
```

Comment:
`NumberShower` would get in the way. I assumed it could be replaced with a string version too, but then I remembered that we have `10^n` notation with a superscript.

I think `text-overflow: ellipsis` would work for inline elements too, or we could look into rendering superscripts with Unicode, but this could get complicated.

Reply:
I think that the number component's React component is pretty nice, I'm hesitant to give that up. The others seem like just strings.

Reply:
Good point about the CSS, will try that.

Reply:
I couldn't get this to work with ~10 min of messing with it. Changed it to 20 chars for now.